### ISSUE TITLE
Ending period detected in first sentence of Javadoc when it is not at the end of the sentence

### ISSUE THREAD

**Issue by patchwork01** [checkstyle/checkstyle#14750]
I have read check documentation: https://checkstyle.org/checks/javadoc/summaryjavadoc.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac TestCheckstyleJavadocSummary.java

/var/tmp $ cat config.xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="SummaryJavadoc"/>
    </module>
</module>

/var/tmp $ cat TestCheckstyleJavadocSummary.java
public class TestCheckstyleJavadocSummary {

    /**
     * Summary sentence with version number 1.0 in the middle but no period at the end
     */
    void foo1() {
    }
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"

/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.14.2-all.jar -c config.xml TestCheckstyleJavadocSummary.java
Starting audit...
Audit done.
```

---

The given input should fail the check as there is no period at the end of the sentence. The check can find the period that ends a sentence by checking if there's a whitespace character after the period.

---

This bug was found while addressing the following issue:
- https://github.com/checkstyle/checkstyle/issues/14689

While fixing that, I also fixed this.
